### PR TITLE
feat(articles): add state and visibility fields

### PR DESCRIPTION
**Pull request by vojtaholik**
image

gif

### CODE REVIEW

File: apps/course-builder-web/src/app/articles/_components/edit-article-form.tsx [hunk 1]
```diff
@@ -131,9 +148,58 @@ export function EditArticleForm({article}: {article: Article}) {
...
+                      </FormControl>
+                      <SelectContent className="">
+                        {ArticleStateSchema.options.map((option) => {
+                          const value = option._def.value
+                          return (
+                            <SelectItem key={value} value={value}>
+                              {value}
+                            </SelectItem>
+                          )
+                        })}
```
Comment (vojtaholik) [inline apps/course-builder-web/src/app/articles/_components/edit-article-form.tsx:182]: schema driven ™

File: apps/course-builder-web/src/trpc/api/routers/articles.ts [hunk 2]
```diff
@@ -86,27 +87,21 @@ export const articlesRouter = createTRPCRouter({
...
     }),
   update: protectedProcedure
-    .input(
-      z.object({
-        articleId: z.string(),
-        title: z.string(),
-        body: z.string().optional().nullable(),
-      }),
-    )
+    .input(ArticleFormSchema)
```
Comment (vojtaholik) [inline apps/course-builder-web/src/trpc/api/routers/articles.ts:90]: this can just take `ArticleFormSchema` to consolidate the types

Comment (joelhooks) [reply apps/course-builder-web/src/trpc/api/routers/articles.ts:90]: that's probably an indicator that `ArticleFormSchema` could be promoted to `ArticleSchema`

Comment (joelhooks) [reply apps/course-builder-web/src/trpc/api/routers/articles.ts:90]: actually that already exists and should be used: https://github.com/badass-courses/course-builder/blob/57/apps/course-builder-web/src/lib/articles.ts#L17-L27

Comment (joelhooks) [reply apps/course-builder-web/src/trpc/api/routers/articles.ts:90]: ahh I see because we have extra validations

**Review by joelhooks — APPROVED** [review]

**Pull request merged**